reward/staking_manager.go

```diff
@@ -109,36 +144,51 @@ func (sm *StakingManager) updateStakingInfo(blockNum uint64) (*StakingInfo, erro
 }
 
 // Subscribe setups a channel to listen chain head event and starts a goroutine to update staking cache.
-func (sm *StakingManager) Subscribe() {
-	sm.chainHeadSub = sm.blockchain.SubscribeChainHeadEvent(sm.chainHeadChan)
+func SubscribeStakingManager() {
```

Comment:
I think this function name is not good.
This function subscribe the chain head event.
How about `SubscribeChainHeadEvent`?

Reply:
As the function doesn't have receiver, putting `StakingManager` in function name is a good idea. How about `StakingManagerSubscribeChainHeadEvent`?

Reply:
It's too long.. How about `StakingManagerSubscribe`?
I am not familiar with singletone pattern.

Reply:
I changed to `StakingManagerSubscribe`.